{"url": "https://api.github.com/repos/rust-lang/rust/issues/19825", "repository_url": "https://api.github.com/repos/rust-lang/rust", "labels_url": "https://api.github.com/repos/rust-lang/rust/issues/19825/labels{/name}", "comments_url": "https://api.github.com/repos/rust-lang/rust/issues/19825/comments", "events_url": "https://api.github.com/repos/rust-lang/rust/issues/19825/events", "html_url": "https://github.com/rust-lang/rust/issues/19825", "id": 51904786, "node_id": "MDU6SXNzdWU1MTkwNDc4Ng==", "number": 19825, "title": "Rust fails to compile on OS X", "user": {"login": "ucarion", "id": 2180153, "node_id": "MDQ6VXNlcjIxODAxNTM=", "avatar_url": "https://avatars.githubusercontent.com/u/2180153?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ucarion", "html_url": "https://github.com/ucarion", "followers_url": "https://api.github.com/users/ucarion/followers", "following_url": "https://api.github.com/users/ucarion/following{/other_user}", "gists_url": "https://api.github.com/users/ucarion/gists{/gist_id}", "starred_url": "https://api.github.com/users/ucarion/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ucarion/subscriptions", "organizations_url": "https://api.github.com/users/ucarion/orgs", "repos_url": "https://api.github.com/users/ucarion/repos", "events_url": "https://api.github.com/users/ucarion/events{/privacy}", "received_events_url": "https://api.github.com/users/ucarion/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 3, "created_at": "2014-12-14T00:47:43Z", "updated_at": "2014-12-14T03:44:19Z", "closed_at": "2014-12-14T02:23:44Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "I just tried compiling 567b90ff095076054c98fa2f08d6c552ae60968d on Mac OS X v10.10 and I got the following error:\n\n```\nerror: internal compiler error: unexpected panic\nnote: the compiler unexpectedly panicked. this is a bug.\nnote: we would appreciate a bug report: http://doc.rust-lang.org/complement-bugreport.html\nnote: run with `RUST_BACKTRACE=1` for a backtrace\ntask 'rustc' panicked at 'assertion failed: end <= self.len()', /Users/ulyssecarion/rust/rust/src/libcore/slice.rs:432\n\nstack backtrace:\n   1:        0x10ccf411f - rt::backtrace::imp::write::h1f139311af8be11d3Ux\n   2:        0x10ccf6518 - failure::on_fail::ha66e7d2ded5b6a30Hhy\n   3:        0x10cf30bf5 - unwind::begin_unwind_inner::h67608f0f28482324FBc\n   4:        0x10cf30903 - unwind::begin_unwind_fmt::h6d059fc061172e6c2yc\n   5:        0x10cf306e2 - rust_begin_unwind\n   6:        0x10cf7f55c - panicking::panic_fmt::h2ddccbd9740c9ebeStl\n   7:        0x10cf79fe2 - panicking::panic::h260960f9ef562752irl\n   8:        0x10a711018 - metadata::loader::Context<'a>::extract_one::h586eca4b58e295edguq\n   9:        0x10a70aeb5 - metadata::loader::Context<'a>::find_library_crate::h8b96d0f1be67645cCkq\n  10:        0x10a701ac0 - metadata::creader::resolve_crate::h4464af585cac84eaZ3o\n  11:        0x10a6fc702 - metadata::creader::Env<'a>.visit..Visitor<'v>::visit_view_item::h5cc85956391d6dabUwo\n  12:        0x10a6fbe0f - metadata::creader::read_crates::h9c23bfae99509c8fOvo\n  13:        0x109a923af - util::common::time::h10367991460009830797\n  14:        0x109a569a0 - driver::phase_3_run_analysis_passes::h7194150d08797315Eta\n  15:        0x109a49727 - driver::compile_input::hdb3965e9a2917589rba\n  16:        0x109adf54a - run::closure.21495\n  17:        0x109aeb8ee - task::TaskBuilder::try_future::closure.22949\n  18:        0x10ccd531d - task::TaskBuilder::spawn_internal::closure.30628\n  19:        0x10cf2ee72 - task::Task::spawn::closure.5565\n  20:        0x10cf9721c - rust_try_inner\n  21:        0x10cf97206 - rust_try\n  22:        0x10cf2ed8f - task::Task::run::h5773b3f802a6e21dfIb\n  23:        0x10cf2eafd - task::Task::spawn::closure.5541\n  24:        0x10cf2febe - thread::thread_start::h5484f4a660df3b96wZb\n  25:     0x7fff8a75d2fc - _pthread_body\n  26:     0x7fff8a75d279 - _pthread_body\n\nmake: *** [x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/stamp.rustc_borrowck] Error 101\n```\n", "closed_by": {"login": "alexcrichton", "id": 64996, "node_id": "MDQ6VXNlcjY0OTk2", "avatar_url": "https://avatars.githubusercontent.com/u/64996?v=4", "gravatar_id": "", "url": "https://api.github.com/users/alexcrichton", "html_url": "https://github.com/alexcrichton", "followers_url": "https://api.github.com/users/alexcrichton/followers", "following_url": "https://api.github.com/users/alexcrichton/following{/other_user}", "gists_url": "https://api.github.com/users/alexcrichton/gists{/gist_id}", "starred_url": "https://api.github.com/users/alexcrichton/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/alexcrichton/subscriptions", "organizations_url": "https://api.github.com/users/alexcrichton/orgs", "repos_url": "https://api.github.com/users/alexcrichton/repos", "events_url": "https://api.github.com/users/alexcrichton/events{/privacy}", "received_events_url": "https://api.github.com/users/alexcrichton/received_events", "type": "User", "site_admin": false}, "reactions": {"url": "https://api.github.com/repos/rust-lang/rust/issues/19825/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/rust-lang/rust/issues/19825/timeline", "performed_via_github_app": null, "state_reason": "completed"}